stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -638,22 +638,43 @@ public class MessageListViewModel(
                 ).enqueue { result ->
                     if (result.isSuccess) {
                         val message = result.data()
+
                         message.attachments.removeAll { attachment ->
-                            if (attachmentToBeDeleted.assetUrl != null) {
-                                attachment.assetUrl == attachmentToBeDeleted.assetUrl
-                            } else {
-                                attachment.imageUrl == attachmentToBeDeleted.imageUrl
+                            val imageUrl = attachmentToBeDeleted.imageUrl
+                            val assetUrl = attachmentToBeDeleted.assetUrl
+
+                            when {
+                                assetUrl != null -> {
+                                    attachment.assetUrl?.substringBefore("?") ==
```

Comment:
substringBefore() will return the original string if it cannot satisfy this condition.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/substring-before.html